src/lib/Microsoft.Health.Fhir.Ingest/Data/EventDataWithJsonBodyToJTokenConverter.cs

```diff
@@ -15,12 +17,20 @@ public class EventDataWithJsonBodyToJTokenConverter : IConverter<EventData, JTok
         public JToken Convert(EventData input)
         {
             EnsureArg.IsNotNull(input, nameof(input));
+            var body = null as JToken;
 
-            var body = input.Body.Count > 0
-                ? JToken.Parse(Encoding.UTF8.GetString(input.Body.Array, input.Body.Offset, input.Body.Count))
-                : null;
-            var data = new { Body = body, input.Properties, input.SystemProperties };
-            var token = JToken.FromObject(data);
+            if (input.Body.Count > 0)
+            {
+                using (var stream = new MemoryStream(input.Body.Array))
+                using (StreamReader sr = new StreamReader(stream, Encoding.UTF8))
+                using (JsonReader reader = new JsonTextReader(sr))
+                {
+                    body = JToken.ReadFrom(reader);
+                }
+            }
+
+            var data = new { Body = body, input.Properties, input.SystemProperties };
```

Comment:
From my testing this actually adds quite a bit of overhead.  In my branch I tweaked it to the following:

```
        private const string BodyAttr = "Body";
        private const string PropertiesAttr = "Properties";
        private const string SystemPropertiesAttr = "SystemProperties";

        private readonly JsonSerializer jsonSerializer = JsonSerializer.CreateDefault();

        public JToken Convert(IEventMessage input)
        {
            EnsureArg.IsNotNull(input, nameof(input));

            JToken token = new JObject();
            JToken body = null;

            if (input.Body.Length > 0)
            {
                using StreamReader streamReader = new StreamReader(input.Body.AsStream(), Encoding.UTF8);
                using JsonReader jsonReader = new JsonTextReader(streamReader);
                body = JToken.ReadFrom(jsonReader);
            }

            token[BodyAttr] = body;
            token[PropertiesAttr] = JToken.FromObject(input.Properties, jsonSerializer);
            token[SystemPropertiesAttr] = JToken.FromObject(input.SystemProperties, jsonSerializer);

            return token;
        }
```

The stream processing will work a little differently in my example (I updated the code to have the original ReadOnlyMemory<byte> and I am taking advantage of that) but the rest should be achievable in this version.  I think the biggest cost is building the new JToken from the object created in line 32.  Starting with an empty JObject and adding the properties to it manually resulted in a ~ 35% increase in number of bytes processed per minute.  I am still not applying the templates but the increase is promising.